application/comit_node/src/settings/mod.rs

```diff
@@ -5,76 +5,165 @@ use crate::seed::Seed;
+    #[serde(default = "default_log_levels")]
+    pub log_levels: LogLevels,
+}
+
+impl Default for ComitNodeSettings {
+    fn default() -> Self {
+        let listen = "/ip4/0.0.0.0/tcp/8011".parse().unwrap();
+        let url = Url::parse("http://localhost:8001").unwrap();
+        let data = rand::thread_rng().gen::<[u8; 32]>();
+
+        ComitNodeSettings {
+            comit: Comit {
+                secret_seed: Seed::from(data),
+            },
+            network: Network {
+                listen: vec![listen],
+            },
+            http_api: HttpSocket {
+                address: IpAddr::V4(Ipv4Addr::LOCALHOST),
+                port: 8000,
+            },
+            btsieve: Btsieve {
+                url,
+                bitcoin: PollParameters {
+                    poll_interval_secs: Duration::from_secs(300),
+                    network: "regtest".into(),
+                },
+                ethereum: PollParameters {
+                    poll_interval_secs: Duration::from_secs(20),
+                    network: "regtest".into(),
+                },
+            },
+            web_gui: Some(HttpSocket {
+                address: IpAddr::V4(Ipv4Addr::LOCALHOST),
+                port: 8080,
+            }),
+            log_levels: LogLevels {
+                comit_node: LevelFilter::Debug,
+            },
+        }
+    }
 }
 
 fn default_log() -> LevelFilter {
     LevelFilter::Debug
 }
 
-#[derive(Clone, Debug, Deserialize)]
+fn default_log_levels() -> LogLevels {
+    LogLevels {
+        comit_node: default_log(),
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+pub struct LogLevels {
+    #[serde(with = "self::serde_log", default = "default_log")]
+    pub comit_node: LevelFilter,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
 pub struct Comit {
     pub secret_seed: Seed,
 }
 
-#[derive(Clone, Debug, Deserialize)]
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
 pub struct Network {
     pub listen: Vec<Multiaddr>,
 }
 
-#[derive(Clone, Debug, Deserialize)]
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
 pub struct HttpSocket {
     pub address: IpAddr,
     pub port: u16,
 }
 
-#[derive(Clone, Debug, Deserialize)]
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
 pub struct Btsieve {
     #[serde(with = "url_serde")]
     pub url: url::Url,
     pub bitcoin: PollParameters,
     pub ethereum: PollParameters,
 }
 
-#[derive(Clone, Debug, Deserialize)]
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
 pub struct PollParameters {
     #[serde(with = "self::serde_duration")]
     pub poll_interval_secs: Duration,
     pub network: String,
 }
 
 impl ComitNodeSettings {
-    pub fn create<D: AsRef<OsStr>, R: AsRef<OsStr>>(
-        default_config: D,
-        run_mode_config: R,
-    ) -> Result<Self, ConfigError> {
+    pub fn create_with_default(path: PathBuf, file_name: &str) -> Result<Self, ConfigError> {
```

Comment:
That's what I tried before, but the problem is, that if the folder does not exist, creating that file will fail, that's why I split it up.

Check a few lines down: 
* `if !path.exists()` --> the folders will be created
* `if default_config.exists()` --> the config file will be created

Reply:
You could do `.ends_with` and check if the passed path is actually the config file and then use `.parent` to get the folder path.

Reply:
It seems like it should be possible to get directory a file is in?

Reply:
I like the current approach more unless we are fine with this function making assumptions about either config file name+type (this should be passed in from the outside). Otherwise you would need to pass in the file name+path anyways leaving us with the same situation.

Reply:
@LLFourn 
> It seems like it should be possible to get directory a file is in?

I would not know how.

Reply:
updated